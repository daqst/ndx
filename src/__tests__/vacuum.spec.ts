import { createIndex, addDocumentToIndex, removeDocumentFromIndex, vacuumIndex } from "..";

const tokenizer = (s: string) => s.split(" ");
const filter = (s: string) => s;

test("one simple document", () => {
  const idx = createIndex(1);
  const removed = new Set();
  const docs = [
    { id: 1, text: "a" },
  ];
  docs.forEach((doc) => {
    addDocumentToIndex(idx, [(d) => d.text], tokenizer, filter, doc.id, doc);
  });
  removeDocumentFromIndex(idx, removed, 1);
  vacuumIndex(idx, removed);

  expect(idx).toMatchSnapshot();
});

test("two documents with shared terms", () => {
  const idx = createIndex(1);
  const removed = new Set();
  const docs = [
    { id: 1, text: "a b c" },
    { id: 2, text: "b c d" },
  ];
  docs.forEach((doc) => {
    addDocumentToIndex(idx, [(d) => d.text], tokenizer, filter, doc.id, doc);
  });
  removeDocumentFromIndex(idx, removed, 1);
  vacuumIndex(idx, removed);

  expect(idx).toMatchSnapshot();
});

test("remove the same document twice", () => {
  const idx = createIndex(1);
  const removed = new Set();
  const docs = [
    { id: 1, text: "a b c" },
    { id: 2, text: "b c d" },
  ];
  docs.forEach((doc) => {
    addDocumentToIndex(idx, [(d) => d.text], tokenizer, filter, doc.id, doc);
  });
  removeDocumentFromIndex(idx, removed, 1);
  removeDocumentFromIndex(idx, removed, 1);
  vacuumIndex(idx, removed);

  expect(idx).toMatchSnapshot();
});

test("two documents with undefined fields", () => {
  const idx = createIndex(2);
  const removed = new Set();
  const docs = [
    { id: 1, text1: "a b c" },
    { id: 2, text1: "b c d" },
  ];
  docs.forEach((doc) => {
    addDocumentToIndex(idx, [(d) => d.text1, (d) => (d as any).text2], tokenizer, filter, doc.id, doc);
  });
  removeDocumentFromIndex(idx, removed, 1);
  vacuumIndex(idx, removed);

  expect(idx).toMatchSnapshot();
});
