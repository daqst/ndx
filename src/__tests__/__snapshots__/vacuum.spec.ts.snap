// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remove the same document twice 1`] = `
Object {
  "docs": Map {
    2 => Object {
      "fieldLengths": Array [
        3,
      ],
      "key": 2,
    },
  },
  "fields": Array [
    Object {
      "avg": 3,
      "sum": 3,
    },
  ],
  "root": Object {
    "charCode": 0,
    "firstChild": Object {
      "charCode": 100,
      "firstChild": null,
      "firstPosting": Object {
        "details": Object {
          "fieldLengths": Array [
            3,
          ],
          "key": 2,
        },
        "next": null,
        "termFrequency": Array [
          1,
        ],
      },
      "next": Object {
        "charCode": 99,
        "firstChild": null,
        "firstPosting": Object {
          "details": Object {
            "fieldLengths": Array [
              3,
            ],
            "key": 2,
          },
          "next": null,
          "termFrequency": Array [
            1,
          ],
        },
        "next": Object {
          "charCode": 98,
          "firstChild": null,
          "firstPosting": Object {
            "details": Object {
              "fieldLengths": Array [
                3,
              ],
              "key": 2,
            },
            "next": null,
            "termFrequency": Array [
              1,
            ],
          },
          "next": Object {
            "charCode": 97,
            "firstChild": null,
            "firstPosting": null,
            "next": null,
          },
        },
      },
    },
    "firstPosting": null,
    "next": null,
  },
}
`;

exports[`two documents with shared terms 1`] = `
Object {
  "docs": Map {
    2 => Object {
      "fieldLengths": Array [
        3,
      ],
      "key": 2,
    },
  },
  "fields": Array [
    Object {
      "avg": 3,
      "sum": 3,
    },
  ],
  "root": Object {
    "charCode": 0,
    "firstChild": Object {
      "charCode": 100,
      "firstChild": null,
      "firstPosting": Object {
        "details": Object {
          "fieldLengths": Array [
            3,
          ],
          "key": 2,
        },
        "next": null,
        "termFrequency": Array [
          1,
        ],
      },
      "next": Object {
        "charCode": 99,
        "firstChild": null,
        "firstPosting": Object {
          "details": Object {
            "fieldLengths": Array [
              3,
            ],
            "key": 2,
          },
          "next": null,
          "termFrequency": Array [
            1,
          ],
        },
        "next": Object {
          "charCode": 98,
          "firstChild": null,
          "firstPosting": Object {
            "details": Object {
              "fieldLengths": Array [
                3,
              ],
              "key": 2,
            },
            "next": null,
            "termFrequency": Array [
              1,
            ],
          },
          "next": Object {
            "charCode": 97,
            "firstChild": null,
            "firstPosting": null,
            "next": null,
          },
        },
      },
    },
    "firstPosting": null,
    "next": null,
  },
}
`;

exports[`two documents with undefined fields 1`] = `
Object {
  "docs": Map {
    2 => Object {
      "fieldLengths": Array [
        3,
        0,
      ],
      "key": 2,
    },
  },
  "fields": Array [
    Object {
      "avg": 3,
      "sum": 3,
    },
    Object {
      "avg": 0,
      "sum": 0,
    },
  ],
  "root": Object {
    "charCode": 0,
    "firstChild": Object {
      "charCode": 100,
      "firstChild": null,
      "firstPosting": Object {
        "details": Object {
          "fieldLengths": Array [
            3,
            0,
          ],
          "key": 2,
        },
        "next": null,
        "termFrequency": Array [
          1,
          0,
        ],
      },
      "next": Object {
        "charCode": 99,
        "firstChild": null,
        "firstPosting": Object {
          "details": Object {
            "fieldLengths": Array [
              3,
              0,
            ],
            "key": 2,
          },
          "next": null,
          "termFrequency": Array [
            1,
            0,
          ],
        },
        "next": Object {
          "charCode": 98,
          "firstChild": null,
          "firstPosting": Object {
            "details": Object {
              "fieldLengths": Array [
                3,
                0,
              ],
              "key": 2,
            },
            "next": null,
            "termFrequency": Array [
              1,
              0,
            ],
          },
          "next": Object {
            "charCode": 97,
            "firstChild": null,
            "firstPosting": null,
            "next": null,
          },
        },
      },
    },
    "firstPosting": null,
    "next": null,
  },
}
`;
